### PR TITLE
Restore JCasC compatibility for `JNLPLauncher.tunnel`

### PR DESCRIPTION
**Pull request by jglick**
#8762 seems to have broken JCasC compatibility for agent definitions using `tunnel`, which is probably unusual but happened to be used by an integration test in CloudBees CI.

I suspect the reason is that `DataBoundConfigurator` is buggy and does not support `@DataBoundSetter` on fields, which despite the name is supported by Jenkins core: it never actually looks for this annotation, and perhaps just assumes that getter/setter pairs not covered by the `@DataBoundConstructor` are properties.

### Testing done

https://github.com/jenkinsci/configuration-as-code-plugin/pull/2441

Removal of `@Deprecated` seems to only be necessary for compatibility with CloudBees CI, which has its own CasC reflection code not based on JCasC and which appears to check for the `@Deprecated` annotation on fields rather than on setters. (For @cloudbees only: verified running selected tests in the `com.cloudbees.casc.server.items` package.)

### Proposed changelog entries

- The `tunnel` property on an `inbound` agent was inadvertently broken for JCasC usage in 2.437. It remains deprecated and usages should be deleted (regression in 2.437).

### Proposed upgrade guidelines

N/A

### Maintainer checklist
- [x] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [x] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [x] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [x] Proper changelog labels are set so that the changelog can be generated automatically.
- [x] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [x] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

### CODE REVIEW

**Review by timja — DISMISSED** [review]

**Review by MarkEWaite — DISMISSED** [review]

**Comment by MarkEWaite** [comment]
This PR is now ready for merge.  We will merge it after approximately 24 hours if there is no negative feedback.

**Review by MarkEWaite — APPROVED** [review]

**Comment by jglick** [comment]
`java.lang.OutOfMemoryError: Java heap space` on `jenkins-war` again

**Comment by MarkEWaite** [comment]
> `java.lang.OutOfMemoryError: Java heap space` on `jenkins-war` again

Reported to the infra help desk as:

* https://github.com/jenkins-infra/helpdesk/issues/3874

**Review by NotMyFault — APPROVED** [review]

**Comment by timja** [comment]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

File: core/src/main/java/hudson/slaves/JNLPLauncher.java
Comment (jglick) [inline core/src/main/java/hudson/slaves/JNLPLauncher.java:153]: This has the unfortunate effect that it is hard to notice improper usages of this method from plugins (for example does not show up as an IDE warning). I wonder if it would be better to just clean up and let anyone with an outdated JCasC file decide whether to clean it up too or tolerate deprecations: https://github.com/jenkinsci/jenkins/pull/9903#issuecomment-2640057543

**Pull request merged**
